auth/auth.go

```diff
@@ -3,16 +3,19 @@ package auth
 import (
 	"errors"
 	"github.com/Frontman-Labs/frontman/config"
+	"net/http"
 )
 
 type TokenValidator interface {
-	ValidateToken(tokenString string) (map[string]interface{}, error)
+	ValidateToken(request *http.Request) (map[string]interface{}, error)
```

Comment:
Passing the request to the validate token method now to utilize the request BasicAuth methods. Also this may be useful when doing APIkey auth, allowing us to configure api-key header